Remotion/Data/DomainObjects.UnitTests/Mapping/MappingConfigurationExtensionsTest.cs

```diff
@@ -0,0 +1,170 @@
+      mappingConfigurationMock.Setup(e => e.GetTypeDefinition(typeof(IDummyInterface), It.IsAny<Func<Type, Exception>>())).Returns(typeDefinition);
+
+      var result = mappingConfigurationMock.Object.GetTypeDefinition(typeof(IDummyInterface));
+      Assert.That(result, Is.SameAs(typeDefinition));
+
+      mappingConfigurationMock.VerifyAll();
+    }
+
+    [Test]
+    public void GetTypeDefinition_WithNonExistentType_Throws ()
+    {
+      Assert.That(
+          () => MappingConfiguration.Current.GetTypeDefinition(typeof(IDummyInterface)),
+          Throws.TypeOf<MappingException>()
+              .With.Message.EqualTo("Mapping does not contain type 'Remotion.Data.DomainObjects.UnitTests.Mapping.MappingConfigurationExtensionsTest+IDummyInterface'."));
+    }
+
+    [Test]
+    public void ContainsClassDefinition ()
+    {
+      var mappingConfigurationMock = new Mock<IMappingConfiguration>(MockBehavior.Strict);
+      mappingConfigurationMock.Setup(e => e.ContainsTypeDefinition(typeof(DummyClass))).Returns(true);
+
+      Assert.That(mappingConfigurationMock.Object.ContainsClassDefinition(typeof(DummyClass)), Is.True);
+
+      mappingConfigurationMock.VerifyAll();
+    }
+
+    [Test]
+    public void ContainsClassDefinition_NonClassType_ReturnsFalse ()
+    {
+      var mappingConfigurationMock = new Mock<IMappingConfiguration>(MockBehavior.Strict);
+      Assert.That(mappingConfigurationMock.Object.ContainsClassDefinition(typeof(IDummyInterface)), Is.False);
+    }
+
+    [Test]
+    public void GetClassDefinition ()
+    {
+      var classDefinition = ClassDefinitionObjectMother.CreateClassDefinition();
+      var mappingConfigurationMock = new Mock<IMappingConfiguration>(MockBehavior.Strict);
+      mappingConfigurationMock
+          .Setup(e => e.GetTypeDefinition(typeof(DummyClass), It.IsAny<Func<Type, Exception>>()))
+          .Returns(classDefinition);
+
+      var result = mappingConfigurationMock.Object.GetClassDefinition(typeof(DummyClass));
+      Assert.That(result, Is.SameAs(classDefinition));
+
+      mappingConfigurationMock.VerifyAll();
+    }
+
+    [Test]
+    public void GetClassDefinition_WithNonClass_Throws ()
+    {
+      var mappingConfigurationMock = new Mock<IMappingConfiguration>(MockBehavior.Strict);
+
+      Assert.That(
+          () => mappingConfigurationMock.Object.GetClassDefinition(typeof(IDummyInterface)),
+          Throws.TypeOf<ArgumentException>()
+              .With.ArgumentExceptionMessageEqualTo(
+                  "The specified type 'Remotion.Data.DomainObjects.UnitTests.Mapping.MappingConfigurationExtensionsTest+IDummyInterface' must be a class.",
+                  "type"));
+    }
+
+    [Test]
+    public void GetClassDefinition_WithCustomExceptionFactory ()
+    {
+      var typeDefinition = TypeDefinitionObjectMother.CreateClassDefinition();
+      var exceptionFactoryMock = new Mock<Func<Type, Exception>>(MockBehavior.Strict);
```

Comment:
In this case I think that the mock makes sense as we want to ensure that nobody is calling the func. Otherwise we would still to return a value in our dummy func thus complicating the test with unnecessary logic

Reply:
Hmm... I guess I usually just added an Assert.Fail or a throw InvalidOpEx into the func and that was that.

Just from a code style perspective, I tend to avoid the mock these days if the alernative is equally safe but requires less mocks (and often, also less code).